File: DuDoong-Api/src/main/java/band/gosrock/api/auth/service/LogoutUseCase.java
package band.gosrock.api.auth.service;


import band.gosrock.api.config.security.SecurityUtils;
import band.gosrock.common.annotation.UseCase;
import band.gosrock.domain.domains.user.adaptor.RefreshTokenAdaptor;
import lombok.RequiredArgsConstructor;

@UseCase
@RequiredArgsConstructor
public class LogoutUseCase {
    private final RefreshTokenAdaptor refreshTokenAdaptor;

    public void execute() {
        Long currentUserId = SecurityUtils.getCurrentUserId();
        refreshTokenAdaptor.deleteByUserId(currentUserId);
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/auth/service/LogoutUseCase.kt
package band.gosrock.api.auth.service

import band.gosrock.common.annotation.UseCase
import band.gosrock.domain.domains.user.adaptor.RefreshTokenAdaptor

@UseCase
class LogoutUseCase(
    private val refreshTokenAdaptor: RefreshTokenAdaptor
) {

    fun execute(userId: Long) {
        refreshTokenAdaptor.deleteByUserId(userId)
    }
}
